CHA-783: add the Getting Started section (#923)
akeneo/pim-api-docs: content/akeneo-event-platform/getting-started.md

# Getting started

This guide will walk you through the essential steps to set up your environment, create subscriptions, and trigger events.

### Prerequisites

> Before proceeding, ensure that you have an active App (it can be a custom App) or an active connection to an Akeneo PIM.

Reviewer: Is it clear for everyone what an app, a custom app or a connection is? Maybe we should add a link to a page that explains this.

Author: links added

    --data-raw '{
        "grant_type": "password",
        "username": "your_API_username",
        "password": "its_password"
    }'
    ```

   You can check the official [token generation documentation](/documentation/authentication.html#token-generation) for further information.

::: info
ℹ️ Note that the token has a lifespan of one hour.
:::

::: info
🛠 You can also use a custom App if you like. As long as you have a `client_id` alongside an `API token`, you are good to go for the next step
:::

### 2. Create a Subscriber

Once you have a valid PIM API token, you can create a subscriber. A subscriber can be seen as an entity where all of your subscriptions will be attached.

```json
curl --request POST 'https://event.prd.sdk.akeneo.cloud/api/v1/subscriber' \
--header 'X-PIM-URL: THE_TARGET_PIM_URL' \
--header 'X-PIM-TOKEN: YOUR_APP_OR_CONNECTION_TOKEN' \
--header 'X-PIM-CLIENT-ID: YOUR_APP_OR_CONNECTION_CLIENT_ID' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "example subscriber name",
    "contact": {
        "technical_email": "subscriber.test.email@example.com"
    }
}'
```

::: info

> 📌 You need to store the ID in the response payload. We will use it in the next step.

Author: I removed this part

    "status": "active",
    "contact": {
        "technical_email": "subscriber.email@example.com"
    },
    "created_at": "2024-06-27T16:26:00.503422Z",
    "updated_at": "2024-06-27T16:26:00.503422Z"
},
```
:::

### 3. Create a Subscription

With a subscriber in place, the next step is to create a subscription to specify which events you want to receive.

In this example, we will use an HTTPS destination (HTTPS is mandatory for secure communication).

To create a subscription, you will need a destination URL.

::: info
💡 You don’t have to worry about the secret part of the configuration for now, they are used to sign the payload (more information below in the API presentation section)
:::

```json
curl --request POST 'https://event.prd.sdk.akeneo.cloud/api/v1/subscriber/$subscriber_Id}/subscription' \
--header 'X-PIM-URL: THE_TARGET_PIM_URL' \
--header 'X-PIM-TOKEN: YOUR_APP_OR_CONNECTION_TOKEN' \
--header 'X-PIM-CLIENT-ID: YOUR_APP_OR_CONNECTION_CLIENT_ID' \
--header 'Content-Type: application/json' \
--data-raw '{
    "source": "pim",
    "subject": "THE_TARGET_PIM_URL",
    "events": [
        "com.akeneo.pim.v1.product.updated"
    ],
    "type": "https",
    "config": {

>         "url": "YOUR_DESTINATION_URL",

Reviewer: have we added the URL check yet?

---

### Basic Use Case: Setting Up Event Subscriptions

This guide provides a step-by-step walkthrough to help you set up event subscriptions using the Akeneo Event Platform. By the end of this guide, you will have created a connection, authenticated your App, subscribed to events, and triggered an event from your PIM.

### 1. Retrieve your credentials from your targeted PIM

::: info
🛠 For every call to the API, you will need `X-PIM-TOKEN` & `X-PIM-CLIENT-ID` .

In this example, we will create a new `connection` in the PIM and use it to generate an `API token`.
:::

1. **Create a connection in Akeneo PIM:**
   - Navigate to **Connect** > **Connection settings** > **Create**.
   - Fill out the form to create the connection.
   - Note the generated `Client ID`, `Secret`, `Username`, and `Password`.

2. **Set Your Environment Variables:**
   - Define the Client ID, Secret, Username, Password, and Akeneo host URL as environment variables:
    ```bash
   export CLIENT_ID="your-client-id"
   export CLIENT_SECRET="your-client-secret"
   export API_USERNAME="your-API-username"
   export API_PASSWORD="your-API-password"

>    export AKENEO_HOST="http://your-host"

Reviewer 2: ```suggestion
   export TARGET_PIM_URL="https://your-pim.cloud.akeneo.com"
```

    ```
   Replace the placeholders with your actual credentials and host URL.

3. **Encode Your Credentials:**
   - Encode the Client ID and Secret in base64 format, separated by a colon (:):
   ```bash
   export BASE64_ENCODED_CLIENTID_AND_SECRET=$(echo -n "$CLIENT_ID:$CLIENT_SECRET" | base64)
   ```

4. **Retrieve Your API Token:**
   - Make the API call to retrieve your `API token` using the environment variables:

    ```bash

>    curl --request POST "$AKENEO_HOST/api/oauth/v1/token" \

Reviewer 2: ```suggestion
   curl --request POST "$TARGET_PIM_URL/api/oauth/v1/token" \
```

   --header "Content-Type: application/json" \
   --header "Authorization: Basic $BASE64_ENCODED_CLIENTID_AND_SECRET" \
   --data-raw '{
   "grant_type": "password",
   "username": "'"$API_USERNAME"'",
   "password": "'"$API_PASSWORD"'"
   }'

>     ```
>    You can check the official [token generation documentation](/documentation/authentication.html#token-generation) for further information.

Reviewer 2: ```suggestion
    ```

    After creating the subscriber, store the id from the response in an environment variable:
```bash
export PIM_API_TOKEN="..."  # Replace with the actual token from the response
```
       
   You can check the official [token generation documentation](/documentation/authentication.html#token-generation) for further information.
```

::: info
ℹ️ Note that the token has a lifespan of one hour.
:::

::: info
🛠 You can also use a custom App if you like. As long as you have a `client_id` alongside an `API token`, you are good to go for the next step
:::

### 2. Create a Subscriber

Once you have a valid PIM API token, you can create a subscriber. A subscriber can be seen as an entity where all of your subscriptions will be attached.

```bash
curl --request POST 'https://event.prd.sdk.akeneo.cloud/api/v1/subscriber' \
--header "X-PIM-URL: $TARGET_PIM_URL" \

> --header "X-PIM-TOKEN: $APP_OR_CONNECTION_TOKEN" \

Reviewer 2: ```suggestion
--header "X-PIM-TOKEN: $PIM_API_TOKEN" \
```

> --header "X-PIM-CLIENT-ID: $APP_OR_CONNECTION_CLIENT_ID" \

Reviewer 2: ```suggestion
--header "X-PIM-CLIENT-ID: $CLIENT_ID" \
```

    },
    "created_at": "2024-06-27T16:26:00.503422Z",
    "updated_at": "2024-06-27T16:26:00.503422Z"
}
```
After creating the subscriber, store the id from the response in an environment variable:
```bash
export SUBSCRIBER_ID="01905a84-a3b7-766e-a49f-5519c35fa7a0"  # Replace with the actual ID from the response
```
:::

### 3. Create a Subscription

With a subscriber in place, the next step is to create a subscription to specify which events you want to receive.

In this example, we will use an HTTPS destination (HTTPS is mandatory for secure communication).

To create a subscription, you will need a destination URL.

::: info
💡 You don’t have to worry about the secret part of the configuration for now, they are used to sign the payload (more information below in the API presentation section)
:::

```bash
curl --request POST 'https://event.prd.sdk.akeneo.cloud/api/v1/subscriber/$SUBSCRIBER_ID/subscription' \
--header "X-PIM-URL: $TARGET_PIM_URL" \
--header "X-PIM-TOKEN: $APP_OR_CONNECTION_TOKEN" \
--header "X-PIM-CLIENT-ID: $APP_OR_CONNECTION_CLIENT_ID" \
--header 'Content-Type: application/json' \
--data-raw '{
    "source": "pim",

>     "subject": "THE_TARGET_PIM_URL",

Reviewer 2: ```suggestion
    "subject": "$TARGET_PIM_URL",
```

# Getting started

This guide will walk you through the essential steps to set up your environment, create subscriptions, and trigger events.

### Prerequisites

Before proceeding, ensure that you have an active App (it can be a custom App) or an active connection to an Akeneo PIM.

> To learn how to create a connection, see the [Authentication Guide](https://api.akeneo.com/documentation/authentication.html#client-idsecret-generation). If you're setting up a custom app, follow the steps in [this tutorial](https://api.akeneo.com/tutorials/how-to-get-your-app-token.html#) to obtain an App token.

Reviewer 3: You can use relative link here

> To learn how to create a connection, see the [Authentication Guide](https://api.akeneo.com/documentation/authentication.html#client-idsecret-generation). If you're setting up a custom app, follow the steps in [this tutorial](https://api.akeneo.com/tutorials/how-to-get-your-app-token.html#) to obtain an App token.

Reviewer 3: App

---

### Basic Use Case: Setting Up Event Subscriptions

> This guide provides a step-by-step walkthrough to help you set up event subscriptions using the Akeneo Event Platform. By the end of this guide, you will have created a connection, authenticated your app, subscribed to events, and triggered an event from your PIM.

Reviewer 3: ```suggestion
This guide provides a step-by-step walkthrough to help you set up event subscriptions using the Akeneo Event Platform. By the end of this guide, you will have created a connection, authenticated your App, subscribed to events, and triggered an event from your PIM.
```

### 1. Retrieve your credentials from your targeted PIM

> <aside>

Reviewer 3: There is kind of callout in API doc

:::info
 For every call to the API, you will need `X-PIM-TOKEN` & `X-PIM-CLIENT-ID` .
In this example, we will create a new `connection` in the PIM and use it to generate an `API token`.
:::

> 🛠 For every call to the API, you will need `X-PIM-TOKEN` & `X-PIM-CLIENT-ID` .

Reviewer 3: API doc callout

In this example, we will create a new `connection` in the PIM and use it to generate an `API token`.

</aside>

1. **Create a connection in Akeneo PIM:**
   - Navigate to **Connect** > **Connection settings** > **Create**.
   - Fill out the form to create the connection.
   - Note the generated `Client ID`, `Secret`, `Username`, and `Password`.
2. **Generate an API Token:**
   - Encode the `client ID` and `secret` in base64 format, separated by a colon (`:`).

    ```bash
    client_id:secret
    4gm4rnoizp8gskgkk080ssoo80040g44ksowwgw844k44sc00s:5dyvo1z6y34so4ogkgksw88ookoows00cgoc488kcs8wk4c40s
    ```

    ```makefile

>     base64 encoded

Reviewer 2: can we provide/document the bash command which get the correct base64 encoded string because I know there is some issues sometimes with encoding with or without padding. In our case if I remember is "with padding which generate some "=" at the end of the string)

    NGdtNHJub2l6cDhnc2tna2swODBzc29vODAwNDBnNDRrc293d2d3ODQ0azQ0c2MwMHM6NWR5dm8xejZ5MzRzbzRvZ2tna3N3ODhvb2tvb3dzMDBjZ29jNDg4a2NzOHdrNGM0MHM=
    ```

3. **Retrieve Your API Token:**
   - After that, you can make the API call to retrieve your `API token`

    ```json
    curl --request POST http://your-host/api/oauth/v1/token \
    --header "Content-Type: application/json" \

>     --header "Authorization: Basic YOUR_BASE_64_CLIENT_ID_AND_SECRET" \

Reviewer 2: in a "step by step" documentation I'd recommand to rely on cli env var.

in point 2
```bash
export BASE64_ENCODED_CLIENTID_AND_SECRET=...
```

and use it in this point
```
--header "Authorization: Basic $BASE64_ENCODED_CLIENTID_AND_SECRET" 
```

(doing this example in github I might be wrongly writing it, but you got the idea)

Reviewer 2: this way all example in this guide will use env var (use it for pim url, client id too)

    --data-raw '{
        "grant_type": "password",
        "username": "your_API_username",
        "password": "its_password"
    }'
    ```

>    You can check the official [token generation documentation](https://api.akeneo.com/documentation/authentication.html#token-generation) for further information.

Reviewer 3: Relative path

> <aside>

Reviewer 3: API doc callout

ℹ️ Note that the token has a lifespan of one hour.

</aside>

> <aside>

Reviewer 3: API doc callout

🛠 You can also use a custom app if you like. As long as you have a `client_id` alongside an `API token`, you are good to go for the next step

</aside>

### 2. Create a Subscriber

> After authentication, you can create a subscriber. A subscriber defines where events should be sent.

Reviewer 2: ```suggestion
Once you have a valid PIM API token, you can create a subscriber. A subscriber can be seen as an entity where all of your subscriptions will be attached.
```

```json
curl --request POST 'https://event.prd.sdk.akeneo.cloud/api/v1/subscriber' \
--header 'X-PIM-URL: THE_TARGET_PIM_URL' \
--header 'X-PIM-TOKEN: YOUR_APP_OR_CONNECTION_TOKEN' \
--header 'X-PIM-CLIENT-ID: YOUR_APP_OR_CONNECTION_CLIENT_ID' \
--header 'Content-Type: application/json' \
--data-raw '{
    "name": "example subscriber name",
    "contact": {
        "technical_email": "subscriber.test.email@example.com"
    }
}'
```

> <aside>

Reviewer 3: API doc callout

📌 You need to store the ID in the response payload. We will use it in the next step.

Response Example :

```json
{
    "id": "01905a84-a3b7-766e-a49f-5519c35fa7a0",
    "tenant_id": "pim_tenant",
    "client_id": "3444ec1b-058e-4208-9b6c-284f47a7aa17",
    "name": "subscriber name",
    "subject": "https://pim.cloud.akeneo.com",
    "status": "active",
    "contact": {
        "technical_email": "subscriber.email@example.com"
    },
    "created_at": "2024-06-27T16:26:00.503422Z",
    "updated_at": "2024-06-27T16:26:00.503422Z"
},
```

</aside>

### 3. Create a Subscription

With a subscriber in place, the next step is to create a subscription to specify which events you want to receive.

In this example, we will use an HTTPS destination (HTTPS is mandatory for secure communication).

To create a subscription, you will need a destination URL.

> <aside>

Reviewer 3: API doc callout

💡 You don’t have to worry about the secret part of the configuration for now, they are used to sign the payload (more information below in the API presentation section)

</aside>

```json

> curl --request POST 'https://event.prd.sdk.akeneo.cloud/api/v1/subscriber/$subscriber_Id}/subscription' \

Reviewer 2: again use the "bash env var" to automatically populate the subscriber_id to be easy to use in a copy/paste way

--header 'X-PIM-URL: THE_TARGET_PIM_URL' \
--header 'X-PIM-TOKEN: YOUR_APP_OR_CONNECTION_TOKEN' \
--header 'X-PIM-CLIENT-ID: YOUR_APP_OR_CONNECTION_CLIENT_ID' \
--header 'Content-Type: application/json' \
--data-raw '{
    "source": "pim",
    "subject": "THE_TARGET_PIM_URL",
    "events": [
        "com.akeneo.pim.v1.product.updated"
    ],
    "type": "https",
    "config": {
        "url": "YOUR_DESTINATION_URL",
        "secret": {
            "primary": "averysecretprimarysecret",
            "secondary": "you can leave empty here, only use to ease your secret rotation use cases"
        }
    }
}'
```

> <aside>

Reviewer 3: API doc callout

📌 Don’t forget to change the `subscriber_id` in the URL to use the one you created in the precedent step.

</aside>

### 4. Trigger an Event from the PIM

> With your subscription in place, you're ready to trigger the event you subscribed to (in this example, `product_updated`). The simplest way to do this is by updating a product directly within the Akeneo PIM UI. This approach allows you to manually trigger the event and observe the changes in real-time.

Reviewer 2: ```suggestion
With your subscription in place, you're ready to trigger the event you subscribed to (in this example, `com.akeneo.pim.v1.product.updated`). The simplest way to do this is by updating a product directly within the Akeneo PIM UI. This approach allows you to manually trigger the event and observe the changes in real-time.
```

> Alternatively, you can also update a product using the [Akeneo PIM REST API](https://api.akeneo.com/api-reference.html#post_products_uuid) if you prefer an automated approach.

Reviewer 3: Relative path
